boudicca.base/publisher-event-html/src/main/kotlin/base/boudicca/publisher/event/html/service/EventService.kt

```diff
@@ -44,25 +48,33 @@ class EventService @Autowired constructor(
 ) {
     private val formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy 'um' HH:mm 'Uhr'", Locale.GERMAN)
     private val localDateFormatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
+    private val additionalMapFilters = listOf(hasField(SemanticKeys.LOCATION_OSM_ID))
 
     @Throws(EventServiceException::class)
     fun search(searchDTO: SearchDTO): List<Map<String, Any?>> {
-        val events = caller.search(QueryDTO(generateQuery(searchDTO), searchDTO.offset ?: 0))
-        return mapEvents(events)
+        val searchResult = caller.search(QueryDTO(generateQuery(searchDTO), searchDTO.offset ?: 0))
+        return mapEvents(searchResult)
     }
 
-    fun generateQuery(searchDTO: SearchDTO): String {
+    @Throws(EventServiceException::class)
+    fun mapSearch(searchDTO: SearchDTO): MapSearchResultDTO {
+        val searchResultDTO =
+            caller.search(QueryDTO(generateQuery(searchDTO, additionalMapFilters), searchDTO.offset ?: 0, 200))
```

Comment:
what does the 200 mean? flagged as magic-number :)

Reply:
well, it is a magic number, namely the max results we wanna show :D i did a lot here since only 30 on the whole map is not enough. i will add a constant